src/Utilities/FlowAnalysis/Analysis/PropertySetAnalysis/PropertySetAnalysis.PropertySetDataFlowOperationVisitor.cs

```diff
@@ -0,0 +1,198 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.FlowAnalysis;
+using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow;
+using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow.PointsToAnalysis;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Analyzer.Utilities.FlowAnalysis.Analysis.PropertySetAnalysis
+{
+    using PropertySetAnalysisData = IDictionary<AbstractLocation, PropertySetAbstractValue>;
+
+    internal partial class PropertySetAnalysis
+    {
+        /// <summary>
+        /// Operation visitor to flow the location validation values across a given statement in a basic block.
+        /// </summary>
+        private sealed class PropertySetDataFlowOperationVisitor :
+            AbstractLocationDataFlowOperationVisitor<PropertySetAnalysisData, PropertySetAnalysisContext, PropertySetAnalysisResult, PropertySetAbstractValue>
+        {
+            private const int MaxInterproceduralCallChain = 1;
+            private readonly ImmutableDictionary<IInvocationOperation, PropertySetAbstractValue>.Builder _hazardousUsageBuilderOpt;
+            private INamedTypeSymbol DeserializerTypeSymbol;
+
+            public PropertySetDataFlowOperationVisitor(PropertySetAnalysisContext analysisContext)
+                : base(analysisContext)
+            {
+                Debug.Assert(analysisContext.OwningSymbol.Kind == SymbolKind.Method);
+                Debug.Assert(analysisContext.PointsToAnalysisResultOpt != null);
+
+                _hazardousUsageBuilderOpt = ImmutableDictionary.CreateBuilder<IInvocationOperation, PropertySetAbstractValue>();
+
+                this.WellKnownTypeProvider.TryGetKnownType(analysisContext.TypeToTrackMetadataName, out this.DeserializerTypeSymbol);
+            }
+
+            public override int GetHashCode()
+            {
+                return HashUtilities.Combine(_hazardousUsageBuilderOpt?.GetHashCode() ?? 0, base.GetHashCode());
+            }
```

Comment:
I know even I had this for the other analysis, though I don't like it :(. Hoping to clean this up soon so that we only need the analysis context's hash code, not for the operation visitor.